hack/common.sh

```diff
@@ -301,6 +297,18 @@ os::build::internal::build_binaries() {
     local host_platform=$(os::build::host_platform)
     local platform
     for platform in "${platforms[@]}"; do
+      os::build::export_targets_and_binaries "${platform}" "$@"
+
+      local -a nonstatics=()
+      local -a tests=()
+      for binary in "${binaries[@]}"; do
+        if [[ "${binary}" =~ ".test"$ ]]; then
```

Comment:
That's not equivalent -- `'.test'$` will test that `'.test'` is followed by a line end anchor, `'.test$'` will test that there is a literal `'$'` after `'.test'`.

This is pre-existing code so I don't think we ought to change it here, but seeing stuff like this makes me feel like there is something wrong with the higher level logic, that `$binaries` list is clearly two lists in one.